src/util/mod.rs

```diff
@@ -65,6 +65,8 @@ pub(crate) mod statistics;
 pub(crate) mod test_util;
 /// A treadmill implementation.
 pub(crate) mod treadmill;
+/// Fast allocation of large zeroed arrays
+pub(crate) mod zeroed_alloc;
```

Comment:
There is an existing module `util::rust_util` where we put a few stuff that we wish Rust provides could but it does not provide yet. This could be moved to `rust_util`, but I don't have a strong opinion on this. It is okay to keep this as a separate module.

Reply:
I think `rust_util` is a good place for this because it contains things that work around limitations of the Rust language.  I moved `zeroed_alloc` to `util::rust_util::zeroed_alloc`.